includes/Core/Modules/Modules.php

```diff
@@ -1077,6 +1079,62 @@ private function get_rest_routes() {
 					),
 				)
 			),
+			new REST_Route(
+				'core/modules/recover-module',
+				array(
+					array(
+						'methods'             => WP_REST_Server::EDITABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$data = $request['data'];
+							$slug = isset( $data['slug'] ) ? $data['slug'] : '';
+							try {
+								$module = $this->get_module( $slug );
+							} catch ( Exception $e ) {
+								return new WP_Error( 'invalid_module_slug', $e->getMessage(), array( 'status' => 404 ) );
+							}
+
+							if ( ! $module->is_shareable() ) {
+								return new WP_Error( 'module_not_shareable', __( 'Module is not shareable.', 'google-site-kit' ), array( 'status' => 404 ) );
+							}
+
+							if ( ! $this->is_module_recoverable( $module ) ) {
+								return new WP_Error( 'module_not_recoverable', __( 'Module is not recoverable.', 'google-site-kit' ), array( 'status' => 403 ) );
+							}
+
+							$check_access_endpoint = '/' . REST_Routes::REST_ROOT . '/' . self::REST_ENDPOINT_CHECK_ACCESS;
+							$check_access_request = new WP_REST_Request( 'POST', $check_access_endpoint );
+							$check_access_request->set_body_params(
+								array(
+									'data' => array(
+										'slug' => 'analytics',
+									),
+								)
+							);
+							$check_access_response = rest_do_request( $check_access_request );
+
+							if ( is_wp_error( $check_access_response ) ) {
+								return $check_access_response;
+							}
+							if ( ! $check_access_response->data['access'] ) {
+								return new WP_Error( 'module_not_accessible', __( 'Module is not accessible by current user.', 'google-site-kit' ), array( 'status' => 403 ) );
+							}
+
+							$current_user_id = get_current_user_id();
+							$module->get_settings()->merge(
+								array(
+									'ownerID' => $current_user_id,
+								)
+							);
+
+							return new WP_REST_Response( array( 'ownerID' => $current_user_id ) );
```

Comment:
```suggestion
							$current_user_id = get_current_user_id();
							$module->get_settings()->merge(
								array(
									'ownerID' => $current_user_id,
								)
							);

							return new WP_REST_Response( array( 'ownerID' => $current_user_id ) );
```

I know this is how the module settings API works, but it's a bit unintuitive to me seeing a `get_settings()->` and not thinking it's just a getter. I didn't expect this to actually write the settings, so let's add a comment to save future developers like me 😉 

Also, let's DRY it up a bit more.

```suggestion
							$module_updates = array( 'ownerID' => get_current_user_id() );

							// Update the settings with the new owner ID (eg. the user
							// making the request).
							$module->get_settings()->merge( $module_updates );

							return new WP_REST_Response( $module_updates );
```

Reply:
This was also thanks to debugging. Apologies - I was debugging this last week - then had a break from this issue and jumped on it again after a few days!